### ISSUE TITLE
PEX_PATH should unify pex sources in addition to 3rdparty deps

### ISSUE THREAD

**Issue by kwlzn** [pantsbuild/pex#317]
given the attached example pex files, it appears that usage of `PEX_PATH` doesn't respect source file imports from one pex to another:

```
[illuminati tt2]$ PEX_VERBOSE=9 PEX_INTERPRETER=1 PEX_PATH=pex1 python2.7 pex2
pex: Detected PEX_PYTHON, re-exec to /opt/twitter_mde/package/python2.7/current/bin/python2.7
pex: Found site-library: /opt/twitter_mde/package/python2.7/661aa656879f98d1b878042a3f00369a1aed87a249570c188e091f2cbcaaa163/lib/python2.7/site-packages
pex: Found site-library: /opt/twitter_mde/package/python2.7/661aa656879f98d1b878042a3f00369a1aed87a249570c188e091f2cbcaaa163/lib/site-python
pex: Found site-library: /opt/twitter_mde/package/python2.7/current/lib/site-python
pex: Found site-library: /opt/twitter_mde/package/python2.7/current/lib/python2.7/site-packages
pex: Not a tainted path element: /private/tmp/tt2/pex2/.bootstrap
pex: Not a tainted path element: /private/tmp/tt2/pex2
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python27.zip
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/plat-darwin
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/plat-mac
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/plat-mac/lib-scriptpackages
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/lib-tk
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/lib-old
pex: Not a tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/lib-dynload
pex: Tainted path element: /opt/twitter_mde/package/python2.7/current/lib/python2.7/site-packages
pex: Scrubbing from user site: /Users/kwilson/.local/lib/python2.7/site-packages
pex: Scrubbing from site-packages: /opt/twitter_mde/package/python2.7/current/lib/python2.7/site-packages
pex: Scrubbing from path_importer_cache: /Users/kwilson/.local/lib/python2.7/site-packages
pex: Scrubbing from path_importer_cache: /opt/twitter_mde/package/python2.7/current/lib/python2.7/site-packages
pex: Activating PEX virtual environment from /private/tmp/tt2/pex2: 0.2ms                                                    
pex:   Searching dependency cache: /private/tmp/tt2/pex2/.deps: 0.1ms
pex: Activating PEX virtual environment from pex1: 0.1ms                                   
pex:   Searching dependency cache: pex1/.deps: 0.1ms
pex: PYTHONPATH contains:
pex:     /private/tmp/tt2/pex2/.bootstrap
pex:     /private/tmp/tt2/pex2
pex:   * /opt/twitter_mde/package/python2.7/current/lib/python27.zip
pex:     /opt/twitter_mde/package/python2.7/current/lib/python2.7
pex:     /opt/twitter_mde/package/python2.7/current/lib/python2.7/plat-darwin
pex:     /opt/twitter_mde/package/python2.7/current/lib/python2.7/plat-mac
pex:     /opt/twitter_mde/package/python2.7/current/lib/python2.7/plat-mac/lib-scriptpackages
pex:     /opt/twitter_mde/package/python2.7/current/lib/python2.7/lib-tk
pex:   * /opt/twitter_mde/package/python2.7/current/lib/python2.7/lib-old
pex:     /opt/twitter_mde/package/python2.7/current/lib/python2.7/lib-dynload
pex:   * - paths that do not exist or will be imported via zipimport
pex: PEX_INTERPRETER specified, dropping into interpreter
Python 2.7.10 (default, Dec 16 2015, 14:09:45) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from example.hello.main import main
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ImportError: No module named example.hello.main
>>> 
```

note that the `PYTHONPATH` in the debug logging does not contain the `pex1` pex files path; however, manually appending this to sys.path causes the import to work.

the existing `PEX_PATH` handling seems to be specific to 3rdparty deps as handled here: https://github.com/pantsbuild/pex/blob/master/pex/pex.py#L70-L79

but the requisite sys.path manipulation required to import sources from the `PEX_PATH`'d pex itself seems to live in the boilerplate here: https://github.com/pantsbuild/pex/blob/master/pex/pex_builder.py#L39-L40

so it seems like we'd probably want to extract a PEX._process_pex_paths() method that would provide a list of additional envs and sys.path components; then make PEX._activate() take an `extra_envs` param and PEX.patch_sys() take an `extra_paths` param; then in PEX.execute() call _process_pex_paths() early on and plumb the extras to _activate() and _patch_sys(). or something along those lines.

this is a blocker for @benjyw 's work on the improved python product pipeline in pants.

**Comment by kwlzn** [comment]
[pexstuff.zip](https://github.com/pantsbuild/pex/files/605222/pexstuff.zip)

**Comment by kwlzn** [comment]
this was fixed in #329
